### ISSUE TITLE
Code generation failure when mapping Q# namespaces to C# namespaces 

### ISSUE THREAD

**Issue by vadym-kl** [microsoft/qsharp-runtime#46]
## Describe the bug
Consider two files `A.B.qs` and `B.qs` with namespaces `A.B` and `B` and operation `Foo` defined in `B.qs`. Fully qualified call `B.Foo()` in `A.B.qs` is valid Q# code that leads to invalid C# code, 
because in C# `B.Foo()` is equivalent to `A.B.Foo()` in this case.

## To Reproduce
First file:
```qsharp
namespace Buggy {

    operation Foo() : Unit is Adj {
    }
}
```
Second file:
```
namespace BuggyBuggy.Buggy {

    operation Foo2 () : Unit {
        Buggy.Foo();
    }   
}
```

## Expected behavior
Successful compilation of Q# program.

## Actual behavior
C# compilation error 
```
 error CS0234: The type or namespace name 'Foo' does not exist in the namespace 'BuggyBuggy.Buggy' (are you missing an assembly reference?)
```

## System information
- OS: Windows

**Additional context**
Not critical, the simple workaround is to avoid the situation by naming namespaces differently.

**Comment by filipjelic** [comment]
@vadym-kl  May I take this one? It would be my first dev contribution to MS 😄

**Comment by swernli** [comment]
Fixed by #346
